Misc/NEWS.d/next/Library/2023-09-09-15-08-37.gh-issue-50644.JUAZOh.rst

```diff
@@ -0,0 +1,4 @@
+Attempts to pickle or create a shallow or deep copy of :mod:`codecs` streams
+now raise a TypeError. Previously, stream pickling produced invalid data,
+which attempts to read resulted in a RecursionError, as well as attempts to
+create a copy of the stream.
```

Comment:
Better wording:  Previously, unpickled or copied streams could produce invalid data and attempts to read from them resulted in a RecursionError.

Reply:
Sorry, maybe my English is so bad that it was misunderstood by you. The problem is:

1. Attempt to pickle a stream finished successfully. But the result is incorrect, and attempt to unpickle it fails with RecursionError.
2. Attempt to copy a stream fails with RecursionError.

So there is no unpickled or copied stream. You cannot unpickle or copy.

How to say this in one sentence?

Reply:
How about this:

Previously, both operations were possible, but produced wrong results and eventually failed with a RecursionError.

Reply:
Unfortunately, it is still not completely technically correct, because copying was not possible, and pickling caused a failure on other end. How about this:

> Previously, pickling produced wrong results that eventually caused unpickling to fail with a RecursionError. Copying failed with a RecursionError.

or this:

> Previously, copying failed with a RecursionError, while pickling produced wrong results that eventually caused unpickling to fail with a RecursionError.

Reply:
Both are great. Sorry for the extra rounds 🙂

Reply:
Thank you. I always grateful for corrections.